src/types/ErrorRate.ts

```diff
@@ -0,0 +1,335 @@
+  let result: RequestTolerance[] = [];
+  if (request) {
+    if (tolerances) {
+      for (let tol of Object.values(tolerances)) {
+        let newReqTol: RequestTolerance = { tolerance: tol, requests: {} };
+        generateRateForTolerance(newReqTol, request);
+        result.push(newReqTol);
+      }
+    } else {
+      const emptyRequest: { [key: string]: Rate } = {};
+      for (let protocol of Object.keys(request)) {
+        emptyRequest[protocol] = emptyRate();
+      }
+      let newReqTol: RequestTolerance = { tolerance: undefined, requests: emptyRequest };
+      generateRateForTolerance(newReqTol, request);
+      result.push(newReqTol);
+    }
+  }
+  return result;
+};
+
+const emptyRate = (): Rate => {
+  return { requestRate: 0, errorRate: 0, errorRatio: 0 };
+};
+
+// Graph Edge
+
+export const graphEdgeColor = (edge: DecoratedGraphEdgeData): ThresholdStatus => {
+  if (edge && edge.responses) {
+    const cy = store.getState().graph.summaryData!.summaryTarget as Cy.Core;
+    const source = decoratedNodeData(cy.$id(edge.source));
+    const target = decoratedNodeData(cy.$id(edge.target));
+
+    // We need to check the configuration for item A outbound requests and configuration of B for inbound requests
+    const configSource = getConfig(source.namespace, source[source.nodeType], source.nodeType);
+    const configTarget = getConfig(target.namespace, target[source.nodeType], target.nodeType);
+
+    // If there is not tolerances with this configuration we'll use defaults
+    const tolerancesSource = configSource?.tolerance.filter(tol => checkExpr(tol.direction, 'outbound'));
+    const tolerancesTarget = configTarget?.tolerance.filter(tol => checkExpr(tol.direction, 'inbound'));
+
+    // Calculate aggregate
+    const outboundEdge = aggregateGraph(
+      transformEdgeResponses(edge.responses, edge.hasTraffic ? edge.protocol : '*'),
+      tolerancesSource
+    );
+    const inboundEdge = aggregateGraph(
+      transformEdgeResponses(edge.responses, edge.hasTraffic ? edge.protocol : '*'),
+      tolerancesTarget
+    );
+
+    // Calculate status
+
+    const outboundEdgeStatus = calculateStatusGraph(outboundEdge);
+    const inboundEdgeStatus = calculateStatusGraph(inboundEdge);
+
+    // Keep status with more priority
+    return outboundEdgeStatus.status.status.priority > inboundEdgeStatus.status.status.priority
+      ? outboundEdgeStatus.status
+      : inboundEdgeStatus.status;
+  }
+
+  return {
+    value: RATIO_NA,
+    status: NA
+  };
+};
+
+const transformEdgeResponses = (requests: Responses, protocol: string): RequestType => {
+  const prot: { [key: string]: number } = {};
+  const result: RequestType = {};
+  result[protocol] = prot;
+
+  for (let [code, responseDetail] of Object.entries(requests)) {
+    const percentRate = (responseDetail as ResponseDetail).flags['-'];
+    result[protocol][code] = Number(percentRate);
+  }
```

Comment:
There may be a misunderstanding about the flags.  A single code may have been returned with various flags.  You want to ignore the flags and sum those percentages.  The '-' flag has no significance other than meaning no flags.  So, for example, you could have something like,

```
"200" : {
   "-"     : "80.0",
   "DC"    : "10.0",
   "FI,FD" : "10.0"
}
```

So the flags are irrelevant here, the above would indicate that 200 was return for 100% of the traffic.